example/lib/api_refactoring_main.dart

```diff
@@ -0,0 +1,391 @@
+                    FormBuilderValidators.compose(<FormFieldValidator<String>>[
+                  /// Makes this field required
+                  FormBuilderValidators.required(),
+
+                  /// Ensures the value entered is numeric - with custom error message
+                  FormBuilderValidators.numeric(
+                    errorText: 'La edad debe ser numérica.',
+                  ),
+
+                  /// Sets a maximum value of 70
+                  FormBuilderValidators.max(70),
+
+                  /// Include your own custom `FormFieldValidator` function, if you want
+                  /// Ensures positive values only. We could also have used `FormBuilderValidators.min( 0)` instead
+                  (String? val) {
+                    if (val != null) {
+                      final int? number = int.tryParse(val);
+                      if (number == null) return null;
+                      if (number < 0) return 'We cannot have a negative age';
+                    }
+                    return null;
+                  }
+                ]),
+
+                     */
+              ),
+              TextFormField(
+                decoration:
+                    const InputDecoration(labelText: 'Age (better way to do)'),
+                keyboardType: TextInputType.number,
+                autovalidateMode: AutovalidateMode.always,
+                validator: req(
+                  isNum(
+                    and([
+                      max(70),
+                      (num value) {
+                        if (value < 0) return 'We cannot have a negative age';
+
+                        return null;
+                      },
+                    ]),
+                    isNumMessage: 'La edad debe ser numérica.',
+                  ),
+                ),
+              ),
+
+              // Required Validator
+              TextFormField(
+                decoration: const InputDecoration(
+                  labelText: 'Required Field',
+                  prefixIcon: Icon(Icons.star),
+                ),
+                // validator: FormBuilderValidators.required(),
+                validator: req(null),
+                autofillHints: const <String>[AutofillHints.name],
+                textInputAction: TextInputAction.next,
+                autovalidateMode: AutovalidateMode.always,
+              ),
+              // Numeric Validator
+              TextFormField(
+                decoration: const InputDecoration(
+                  labelText: 'Numeric Field',
+                  prefixIcon: Icon(Icons.numbers),
+                ),
+                keyboardType: TextInputType.number,
+                //validator: FormBuilderValidators.numeric(),
+                // To maintain the previous behavior, we use msg. Below, there is a more precise way of doing that.
+                validator: msg(
+                    FormBuilderLocalizations.current.numericErrorText,
+                    req(isNum(null))),
```

Comment:
# Clarifications
1. msg is only an alias I used for withMessage
2. It replaces any inner message with `errorMessage`. I used it to reproduce the behavior of the current API.
3. It is useful also for changing the message of direct validator implementations, like `isTrue` or `isFalse`. 
4. In the example that you highlighted, without `withMessage`, it would return the isRequired default error message when the input is empty and the isNumeric default error message when the field is not empty but is not numeric. With the `withMessage`, it will return only the isNumeric error message, even if the field is empty.

# Actions
1. I changed the name to: replaceErrorMessage and replaceErrMsg. What do you think?
2. Maybe we can discuss about direct validators in another thread also. They are `String? Function(T value)` validators.

The implementation:
```dart
/// Replaces any inner message with [errorMessage]. It is useful also for changing
/// the message of direct validator implementations, like [isTrue] or [isFalse].
Validator<T> withMessage<T extends Object?>(
  String errorMessage,
  Validator<T> v,
) {
  return (value) {
    final vErrorMessage = v(value);
    if (vErrorMessage != null) {
      return errorMessage;
    }
    return null;
  };
}
```

Reply:
Ok, I understand better now. Make sense.
I think that the name here is the most important for DX. 

- withMessage: means that something inside this method, get a message.
- replaceMessage: means that replace some message

Other options could be:
- singleMessageError: to make reference to a single (one) message error to all validators inside the method
- unifiedMessageError: to make reference to an unified message error for all validators inside the method

I think that some name that make reference to the final action, that is:
- ignore all error messages of validators inside this  method (`withMessage`) ,
- and also show a single message that replace/unified/? the error message for this all validators;

should better.

Reply:
I changed to overrideErrorMsg. I think unify may give the idea that the function processes in some way the inner message, what does not happen, it simply overrides any inner message. But what do you think?

Reply:
Yeah, I like it